packages/webapp/src/elections/components/ongoing-election.tsx

```diff
@@ -1,32 +1,46 @@
-import { useCurrentElection, useMemberStats } from "_app";
-import { Container, Heading, Text } from "_app/ui";
-import { ElectionRoundData, ElectionStatus } from "elections/interfaces";
+import dayjs from "dayjs";
+
+import { useCommunityGlobals, useMemberStats } from "_app";
+import { Container, Heading, Loader, Text } from "_app/ui";
+import { ElectionStatus } from "elections/interfaces";
 
 import * as Ongoing from "./ongoing-election-components";
 
 // TODO: Make sure time zone changes during election are handled properly
-export const OngoingElection = () => {
-    const { data: currentElection } = useCurrentElection();
-    const { data: memberStats } = useMemberStats();
+export const OngoingElection = ({ election }: { election: any }) => {
+    const {
+        data: globals,
+        isLoading: isLoadingGlobals,
+        isError: isErrorGlobals,
+    } = useCommunityGlobals();
+    const {
+        data: memberStats,
+        isLoading: isLoadingMemberStats,
+        isError: isErrorMemberStats,
+    } = useMemberStats();
 
-    if (!currentElection || !memberStats) {
+    const isLoading = isLoadingGlobals || isLoadingMemberStats;
+    if (isLoading) {
         return (
             <Container>
-                <Heading size={2}>Loading</Heading>
+                <Loader />
             </Container>
         );
     }
 
-    let roundData = currentElection as ElectionRoundData;
-    if (currentElection?.electionState === ElectionStatus.Final) {
-        roundData = {
-            electionState: currentElection?.electionState,
-            round: memberStats?.ranks.length,
-            // TODO: Reduce time for sortition round to two hours in contract
-            round_end: currentElection?.seed.end_time,
-        };
+    const isError = isErrorGlobals || isErrorMemberStats;
+    if (isError || !memberStats) {
+        return <Ongoing.ErrorLoadingElection />;
     }
 
+    const { election_round_time_sec, election_break_time_sec } = globals;
+    const roundDurationSec = election_round_time_sec + election_break_time_sec;
+    const roundDurationMs = roundDurationSec * 1000;
+    const roundIndex = election.round ?? memberStats.ranks.length;
+    const roundEndTimeRaw = election.round_end ?? election.seed.end_time;
+    const roundEndTime = dayjs(roundEndTimeRaw + "Z");
+    const roundStartTime = dayjs(roundEndTime).subtract(roundDurationMs);
```

Comment:
This is a good candidate for factoring out.

Reply:
yeah probably create another get dates utility function in the utils.ts returning all of these values in an object. (can be a later pr)